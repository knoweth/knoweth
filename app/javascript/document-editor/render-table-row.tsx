import { ELEMENT_TD, ELEMENT_TR } from "@udecode/slate-plugins";
import { RenderElementProps } from "slate-react";
import React from "react";

export function NoteCell({
  hidden,
  children,
}: {
  hidden: boolean;
  children: React.ReactChild;
}) {
  if (hidden) {
    return (
      <td
        data-cell-hidden="true"
        className="p-3 text-muted bg-light user-select-none text-center"
      >
        Answer hidden
      </td>
    );
  }

  return <td data-cell-hidden="false">{children}</td>;
}

/**
 * Renders a table row with a highlight for rows that are marked to be
 * flashcards
 */
export function renderTrWithCardId({
  attributes,
  ...props
}: RenderElementProps) {
  if (props.element.type === ELEMENT_TR) {
    return (
      <tr
        className={props.element.cardId !== undefined ? "bg-secondary" : ""}
        title={
          props.element.cardId !== undefined
            ? `Flashcard ID: ${props.element.cardId}`
            : undefined
        }
        {...attributes}
        {...props}
      />
    );
  }
}

/**
 * Render a hidden td, e.g. one shown while reviewing
 */
export function renderTdWithOcclusion({
  attributes,
  ...props
}: RenderElementProps) {
  if (props.element.type === ELEMENT_TD) {
    return (
      <NoteCell
        hidden={props.element.hidden == true}
        // Used for auto-scrolling behavior (document scrolls to hidden cell in reviewer)
        children={props.children}
      />
    );
  }
}
