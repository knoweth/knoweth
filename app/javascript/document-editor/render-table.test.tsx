import React from "react";
import { render } from "@testing-library/react";
import { NoteCell } from "./render-table-row";

describe("NoteCell", () => {
  it("hidden cell matches snapshot", () => {
    const { container } = render(
      <table>
        <tbody>
          <tr>
            <NoteCell hidden children={<>test</>} />
          </tr>
        </tbody>
      </table>
    );
    expect(container).toMatchSnapshot();
  });
});
