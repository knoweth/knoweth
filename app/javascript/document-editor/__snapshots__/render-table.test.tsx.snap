// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteCell hidden cell matches snapshot 1`] = `
<div>
  <table>
    <tbody>
      <tr>
        <td
          class="p-3 text-muted bg-light user-select-none text-center"
        >
          Answer hidden
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
